Bug: test_all_deps_importable — onnxruntime incompatible with NumPy 2.x in pipeline Docker image

## What
`test_all_deps_importable` in `pipeline/tests/test_integration.py` fails because `onnxruntime` in the pipeline Docker image was compiled for NumPy 1.x but NumPy 2.x is installed in the image.

## Error output
```
A module that was compiled using NumPy 1.x cannot be run in
NumPy 2.4.4 as it may crash. To support both 1.x and 2.x
versions of NumPy, modules must be compiled with NumPy 2.0.

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python3.11/site-packages/onnxruntime/__init__.py", line 23, in <module>
    from onnxruntime.capi._pybind_state import ExecutionMode  # noqa: F401
  File "/usr/local/lib/python3.11/site-packages/onnxruntime/capi/_pybind_state.py", line 32, in <module>
    from .onnxruntime_pybind11_state import *  # noqa
AttributeError: _ARRAY_API not found

AssertionError: assert 1 == 0
 +  where 1 = CompletedProcess(..., returncode=1).returncode
```

## Expected vs actual
- **Expected:** `docker run ... python -c "import transformers, torch, onnxruntime, ..."` exits 0
- **Actual:** exits 1 — `onnxruntime` fails to import due to NumPy 2.x incompatibility

## Likely implementation file
`pipeline/Dockerfile` — pin `onnxruntime` to a version compatible with NumPy 2.x, or pin `numpy<2`.

## Relevant acceptance criterion
Integration test `test_all_deps_importable` must pass (all pipeline dependencies importable inside the Docker image).

--- a/pipeline/Dockerfile
+++ b/pipeline/Dockerfile
@@ -12,3 +12,5 @@ RUN pip install --no-cache-dir -r requirements.txt
 
 # Copy pipeline scripts; tests are excluded
 COPY *.py ./
+
+CMD []

--- a/pipeline/requirements.txt
+++ b/pipeline/requirements.txt
@@ -1,6 +1,7 @@
 transformers==4.39.3
 torch>=2.0.0
 onnxruntime==1.17.3
+numpy<2
 tritonclient[grpc]
 tqdm
 qdrant-client

